### PR TITLE
Fix a crash due to incorrect condition in `zephir_create_instance_params`

### LINKED ISSUE #1339: create_instance_params seg fault when more than 10 params
create_instance_params is seg faulting when creating class with more than 10 dependencies:

```
#0  0x00007ffff393957b in _zend_is_inconsistent (ht=0xa, file=0x7ffff3f0cb58 "/root/.phpbrew/build/5.6.19-debug/Zend/zend_hash.c", line=1044)
    at /root/.phpbrew/build/5.6.19-debug/Zend/zend_hash.c:60
#1  0x00007ffff393c5e9 in zend_hash_move_forward_ex (ht=0xa, pos=0x7fffffffb568) at /root/.phpbrew/build/5.6.19-debug/Zend/zend_hash.c:1044
#2  0x00007fffdcfdfa44 in zephir_create_instance_params () from /root/.phpbrew/php/5.6.19-debug/lib/php/extensions/debug-non-zts-20131226/instancetest.so
#3  0x00007fffdcffee54 in zim_Instancetest_ExampleService_test () from /root/.phpbrew/php/5.6.19-debug/lib/php/extensions/debug-non-zts-20131226/instancetest.so
#4  0x00007ffff396e6dd in zend_do_fcall_common_helper_SPEC (execute_data=0x555555ceabf0) at /root/.phpbrew/build/5.6.19-debug/Zend/zend_vm_execute.h:558
#5  0x00007ffff396eeae in ZEND_DO_FCALL_BY_NAME_SPEC_HANDLER (execute_data=0x555555ceabf0) at /root/.phpbrew/build/5.6.19-debug/Zend/zend_vm_execute.h:693
#6  0x00007ffff396dd4c in execute_ex (execute_data=0x555555ceabf0) at /root/.phpbrew/build/5.6.19-debug/Zend/zend_vm_execute.h:363
#7  0x00007ffff396ddd5 in zend_execute (op_array=0x555555d23b60) at /root/.phpbrew/build/5.6.19-debug/Zend/zend_vm_execute.h:388
#8  0x00007ffff392914a in zend_execute_scripts (type=8, retval=0x0, file_count=3) at /root/.phpbrew/build/5.6.19-debug/Zend/zend.c:1341
#9  0x00007ffff388f1f3 in php_execute_script (primary_file=0x7fffffffdc30) at /root/.phpbrew/build/5.6.19-debug/main/main.c:2610
#10 0x00007ffff39dc251 in php_handler (r=0x7ffff7e040a0) at /root/.phpbrew/build/5.6.19-debug/sapi/apache2handler/sapi_apache2.c:667
#11 0x00005555555aabe0 in ap_run_handler ()
#12 0x00005555555ab129 in ap_invoke_handler ()
#13 0x00005555555c06ca in ap_process_async_request ()
#14 0x00005555555c09a4 in ap_process_request ()
#15 0x00005555555bd442 in ?? ()
#16 0x00005555555b4220 in ap_run_process_connection ()
#17 0x00007ffff448c767 in ?? () from /usr/lib/apache2/modules/mod_mpm_prefork.so
#18 0x00007ffff448c96c in ?? () from /usr/lib/apache2/modules/mod_mpm_prefork.so
#19 0x00007ffff448d6b1 in ?? () from /usr/lib/apache2/modules/mod_mpm_prefork.so
#20 0x00005555555918ae in ap_run_mpm ()
#21 0x000055555558b046 in main ()
```

There is repo with code for reproduce:

https://github.com/Jurigag/instancetest

And php code:

``` php
$test = \InstanceTest\AnotherService::test();
$test->get();
$test = \InstanceTest\ExampleService::test();
$test->get();
```

First class like AnotherService works fine. But ExampleService is seg faulting.

This issue is related to phalcon issue - https://github.com/phalcon/cphalcon/issues/12299

This seg fault is from php 5.6 and problem is reproduced on it. On PHP 7(debug and builddev) stack looks like this:

```
#0  0x0000555555be2568 in ?? ()
#1  0x0000555555be2588 in ?? ()
#2  0x0000555555be25a8 in ?? ()
#3  0x0000555555be25c8 in ?? ()
#4  0x00007fffffffaca0 in ?? ()
#5  0x00007fffffffb1a0 in ?? ()
#6  0x00007ffff7e111c0 in ?? ()
#7  0x0000555555bdfd98 in ?? ()
#8  0x0000000000000207 in ?? ()
#9  0x00007fff00000c08 in ?? ()
#10 0x00007ffff390a748 in zend_ast_get_lineno (ast=0xc08) at /root/.phpbrew/build/7.0.5-debug/Zend/zend_ast.h:240
```

So it's super useless.
##

### PR DESCRIPTION
**Pull request by sjinks**
This fixes #1339, and includes the test case by @Jurigag from #1427

### CODE REVIEW

File: kernels/ZendEngine3/object.c [hunk 1]
```diff
@@ -1194,7 +1194,7 @@ int zephir_create_instance_params(zval *return_value, const zval *class_name, co
 			zval *item;
 			int i = 0;
 
-			if (likely(param_count) <= 10) {
+			if (EXPECTED(param_count <= 10)) {
```
Comment (sergeyklay) [inline kernels/ZendEngine3/object.c:1197]: Good catch!

Comment (sjinks) [reply kernels/ZendEngine3/object.c:1197]: I have pushed the same fix to ZE2 directly to master to save some time.

Comment (sergeyklay) [reply kernels/ZendEngine3/object.c:1197]: Okay

Comment (sjinks) [reply kernels/ZendEngine3/object.c:1197]: oops, to my master

Comment (Jurigag) [reply kernels/ZendEngine3/object.c:1197]: Well IN ZE2 this fix - https://github.com/phalcon/zephir/pull/1342/files#diff-a10f1751cb10ff71867e53250b0acb36L1560 fixed an issue anyway

So maybe you need to revert it

Comment (sjinks) [reply kernels/ZendEngine3/object.c:1197]: @Jurigag no, your fix is fine, too. Your fix addresses the issue with the way how parameters are passed, my fix addresses the case when there are too many parameters and they corrupt the stack.

Your test case passes 11 parameters, the 11th one possibly overwrites `static_params` variable (it depends on the stack layout and how the compiler arranged local variables). But if you increase the number of parameters, they will eventually overwrite the stack pointer and crash the application.

Actually this is why the issue is compiler-specific: VC does not have __builtin_constant_p therefore the code works. clang could probably lay out the local variables differently, and the app does not crash. But gcc 4.8 does what it does, and the 11th parameter overwrites the stack pointer.

**Comment by Jurigag** [comment]
Well thanks for explaining.

**Pull request merged**